### DISCUSSION TITLE
picard: lookup takes longer without netlink

### DISCUSSION THREAD

**Discussion by haplo** [netblue30/firejail#6543] [Q&A]
I noticed that when running [Musicbrainz's Picard](https://picard.musicbrainz.org/) with firejail I see the following log errors (in `journalctl`):

```
python[18053]: Could not create AF_NETLINK socket (Operation not supported)
```

An easy solution is to add this to `picard.local`:

```
protocol unix,inet,inet6,netlink
ignore protocol
```

However I'm not sure whether Picard really needs netlink, it seems to work without it although sometimes the lookup hangs for a long time, or never finishes and I have to restart.

Any ideas?

**Comment by kmk3** [comment]
> I noticed that when running [Musicbrainz's
> Picard](https://picard.musicbrainz.org/) with firejail I see the following
> log errors (in `journalctl`):
> 
> ```
> python[18053]: Could not create AF_NETLINK socket (Operation not supported)
> ```

This message is common when netlink is blocked.

> An easy solution is to add this to `picard.local`:
> 
> ```
> protocol unix,inet,inet6,netlink
> ignore protocol
> ```
> 
> However I'm not sure whether Picard really needs netlink, it seems to work
> without it although sometimes the lookup hangs for a long time, or never
> finishes and I have to restart.
> 
> Any ideas?

If it causes major issues, then it would be good to add netlink to the protocol
list.

Feel free to open a PR.

Also, please consider reporting this as a bug to picard and linking it here.

For any new profile issues, please follow the bug report template:

* <https://github.com/netblue30/firejail/issues/new?template=bug_report.md>
